ddtrace/contrib/kafka/patch.py

```diff
@@ -149,7 +162,15 @@ def traced_produce(func, instance, args, kwargs):
         span.set_tag_str(COMPONENT, config.kafka.integration_name)
         span.set_tag_str(SPAN_KIND, SpanKind.PRODUCER)
         span.set_tag_str(kafkax.TOPIC, topic)
-        span.set_tag_str(kafkax.MESSAGE_KEY, ensure_text(message_key, errors="replace"))
+
+        if isinstance(instance, _SerializingProducer):
+            if hasattr(instance, "_key_serializer"):
```

Comment:
Using an internal attribute here is a bit brittle. This check can start failing at anytime and it could be painful to debug why the message was not set. 

We should try to use a public attribute if we can, if not we should atleast log a warning when this check fails.

Reply:
I will add the warning log